TestFoundation/FTPServer.swift

```diff
@@ -0,0 +1,274 @@
+    import SwiftXCTest
+#endif
+
+class _FTPSocket {
+
+    private var listenSocket: Int32!
+    private var socketAddress = UnsafeMutablePointer<sockaddr_in>.allocate(capacity: 1)
+    private var socketAddress1 = UnsafeMutablePointer<sockaddr_in>.allocate(capacity: 1)
+    private var connectionSocket: Int32!
+    var dataSocket: Int32! // data socket for communication
+    var dataSocketPort: UInt16! // data socket port, should be sent as part of header
+    private func isNotNegative(r: CInt) -> Bool {
+        return r != -1
+    }
+
+    private func isZero(r: CInt) -> Bool {
+        return r == 0
+    }
+
+    private func attempt(_ name: String, file: String = #file, line: UInt = #line, valid: (CInt) -> Bool,  _ b: @autoclosure () -> CInt) throws -> CInt {
+        let r = b()
+        guard valid(r) else { throw ServerError(operation: name, errno: r, file: file, line: line) }
+        return r
+    }
+
+    init(port: UInt16) throws {
+        #if os(Linux)
+            let SOCKSTREAM = Int32(SOCK_STREAM.rawValue)
+        #else
+            let SOCKSTREAM = SOCK_STREAM
+        #endif
+        listenSocket = try attempt("socket", valid: isNotNegative, socket(AF_INET, SOCKSTREAM, Int32(IPPROTO_TCP)))
+        var on: Int = 1
+        _ = try attempt("setsockopt", valid: isZero, setsockopt(listenSocket, SOL_SOCKET, SO_REUSEADDR, &on, socklen_t(MemoryLayout<Int>.size)))
+        let sa = createSockaddr(port)
+        socketAddress.initialize(to: sa)
+        try socketAddress.withMemoryRebound(to: sockaddr.self, capacity: MemoryLayout<sockaddr>.size, {
+            let addr = UnsafePointer<sockaddr>($0)
+            _ = try attempt("bind", valid: isZero, bind(listenSocket, addr, socklen_t(MemoryLayout<sockaddr>.size)))
+        })
+        
+        dataSocket = try attempt("socket", valid: isNotNegative, socket(AF_INET, SOCKSTREAM, Int32(IPPROTO_TCP)))
+        var on1: Int = 1
+        _ = try attempt("setsockopt", valid: isZero, setsockopt(dataSocket, SOL_SOCKET, SO_REUSEADDR, &on1, socklen_t(MemoryLayout<Int>.size)))
+        let sa1 = createSockaddr(port+1)
+        socketAddress1.initialize(to: sa1)
+        try socketAddress1.withMemoryRebound(to: sockaddr.self, capacity: MemoryLayout<sockaddr>.size, {
+            let addr = UnsafeMutablePointer<sockaddr>($0)
+            _ = try attempt("bind", valid: isZero, bind(dataSocket, addr, socklen_t(MemoryLayout<sockaddr>.size)))
+            var sockLen = socklen_t(MemoryLayout<sockaddr>.size)
+            _ = try attempt("listen", valid: isZero, listen(dataSocket, SOMAXCONN))
+            // Open the data port asynchronously. Port should be opened before ESPV header communication.
+            DispatchQueue(label: "delay").async {
+                do {
+                    self.dataSocket = try self.attempt("accept", valid: self.isNotNegative, accept(self.dataSocket, addr, &sockLen))
+                    self.dataSocketPort = sa1.sin_port
+                } catch {
+                }
```

Comment:
I know this is test code, but do you really mean to eat any error and continue? Seems like the test should abort.

Reply:
Isn't the entire purpose of the test suite to log failures?